pyomo/repn/plugins/lp_writer.py

```diff
@@ -0,0 +1,600 @@
+
+        #
+        # Process objective
+        #
+        if not component_map[Objective]:
+            objectives = [Objective(expr=1)]
+            objectives[0].construct()
+        else:
+            objectives = []
+            for blk in component_map[Objective]:
+                objectives.extend(
+                    blk.component_data_objects(
+                        Objective, active=True, descend_into=False, sort=sorter
+                    )
+                )
+        if len(objectives) > 1:
+            raise ValueError(
+                "More than one active objective defined for input model '%s'; "
+                "Cannot write legal LP file\nObjectives: %s"
+                % (model.name, ' '.join(obj.name for obj in objectives))
+            )
+
+        obj = objectives[0]
+        ostream.write(
+            ("min \n%s:\n" if obj.sense == minimize else "max \n%s:\n")
+            % (getSymbol(obj, labeler),)
+        )
+        repn = objective_visitor.walk_expression(obj.expr)
+        if repn.nonlinear is not None:
+            raise ValueError(
+                f"Model objective ({obj.name}) contains nonlinear terms that "
+                "cannot be written to LP format"
+            )
+        if repn.constant or not (repn.linear or getattr(repn, 'quadratic', None)):
+            # Older versions of CPLEX (including 12.6) and all versions
+            # of GLPK (through 5.0) do not support constants in the
+            # objective in LP format.  To avoid painful bookkeeping, we
+            # introduce the following "variable", constrained to the
+            # value 1.
+            #
+            # In addition, most solvers do no tolerate an empty
+            # objective, this will ensure we at least write out
+            # 0*ONE_VAR_CONSTANT.
+            repn.linear[id(ONE_VAR_CONSTANT)] = repn.constant
+            repn.constant = 0
+        self.write_expression(ostream, repn, True)
+        aliasSymbol(obj, '__default_objective__')
+        if with_debug_timing:
+            timer.toc('Objective %s', obj, level=logging.DEBUG)
+
+        ostream.write("\ns.t.\n")
+
+        #
+        # Tabulate constraints
+        #
+        skip_trivial_constraints = self.config.skip_trivial_constraints
+        have_nontrivial = False
+        last_parent = None
+        for con in ordered_active_constraints(model, self.config):
+            if with_debug_timing and con.parent_component() is not last_parent:
+                timer.toc('Constraint %s', last_parent, level=logging.DEBUG)
+                last_parent = con.parent_component()
+            lb = con.lb
+            ub = con.ub
+            if lb is None and ub is None:
+                # Note: you *cannot* output trivial (unbounded)
+                # constraints in LP format.  I suppose we could add a
+                # slack variable if skip_trivial_constraints is False,
+                # but that seems rather silly.
+                continue
```

Comment:
Should this be ignored? Or maybe it would be polite to at least log something suggesting that there might be a modeling error? I'm just thinking the most common cause of this is a typo, right? Or a bug in a transformation or something.